File: solutions/412.fizz-buzz/fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
const fizzBuzz = n => {
  let res = Array(n)
  for (let i = 1; i <= n; i++) {
    if (i % 3 === 0) {
      if (i % 5 === 0) {
        res[i - 1] = "FizzBuzz"
      } else res[i - 1] = "Fizz"
    } else if (i % 5 === 0) {
      res[i - 1] = "Buzz"
    } else {
      res[i - 1] = String(i)
    }
  }
  return res
}

File: solutions/412.fizz-buzz/fizz-buzz.py
class Solution:
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        ret = []

        for i in range(n):
            cur = i + 1

            if cur % 3 == 0:
                if cur % 5 == 0:
                    ret.append('FizzBuzz')
                else:
                    ret.append('Fizz')
            elif cur % 5 == 0:
                ret.append('Buzz')
            else:
                ret.append(str(cur))

        return ret
